controllers/api_v1/login_test.go

```diff
@@ -0,0 +1,172 @@
+package api_v1_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"testing"
+
+	"github.com/gutakk/go-google-scraper/config"
+	"github.com/gutakk/go-google-scraper/controllers"
+	"github.com/gutakk/go-google-scraper/controllers/api_v1"
+	"github.com/gutakk/go-google-scraper/db"
+	"github.com/gutakk/go-google-scraper/models"
+	"github.com/gutakk/go-google-scraper/oauth"
+	testConfig "github.com/gutakk/go-google-scraper/tests/config"
+	testDB "github.com/gutakk/go-google-scraper/tests/db"
+	testHttp "github.com/gutakk/go-google-scraper/tests/http"
+	"github.com/gutakk/go-google-scraper/tests/oauth_test"
+	"github.com/gutakk/go-google-scraper/tests/path_test"
+
+	"github.com/bxcodec/faker/v3"
+	"github.com/gin-gonic/gin"
+	"github.com/go-oauth2/oauth2/v4/errors"
+	"github.com/stretchr/testify/suite"
+	"golang.org/x/crypto/bcrypt"
+	"gopkg.in/go-playground/assert.v1"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+)
+
+func init() {
+	gin.SetMode(gin.TestMode)
+
+	if err := os.Chdir(path_test.GetRoot()); err != nil {
+		panic(err)
+	}
+
+	config.LoadEnv()
+	_ = oauth.SetupOAuthServer()
+	database, _ := gorm.Open(postgres.Open(testDB.ConstructTestDsn()), &gorm.Config{})
+	db.GetDB = func() *gorm.DB {
+		return database
+	}
+
+	_ = db.GetDB().AutoMigrate(&models.User{})
+}
+
+type LoginAPIControllerDbTestSuite struct {
+	suite.Suite
+	engine      *gin.Engine
+	user        models.User
+	oauthClient oauth_test.OAuthClient
+	headers     http.Header
+}
+
+func (s *LoginAPIControllerDbTestSuite) SetupTest() {
+	s.engine = testConfig.GetRouter(false)
+	new(api_v1.LoginAPIController).ApplyRoutes(controllers.PublicAPIGroup(s.engine.Group("/api/v1")))
+
+	s.headers = http.Header{}
+	s.headers.Set("Content-Type", "application/x-www-form-urlencoded")
+
+	hashedPassword, _ := bcrypt.GenerateFromPassword([]byte("password"), bcrypt.DefaultCost)
```

Comment:
@gutakk Regarding [our RFC](https://github.com/nimblehq/compass/issues/469) we should handle all errors on the test file also 🙌🏼

Reply:
Fixed in 955b2f87a5ce36105f16855acd46bbaa47bc9ad5